sdk/transport_policies/curl/src/az_curl.c

```diff
@@ -291,9 +292,10 @@ static int32_t _az_http_client_curl_upload_read_callback(
 
   int32_t data_length = az_span_length(*mut_span);
   int32_t to_copy = (data_length < curl_size) ? data_length : curl_size;
-  AZ_RETURN_IF_FAILED(
-      az_span_append(*mut_span, az_span_init((uint8_t *)ptr, to_copy, to_copy), mut_span));
-
+  memcpy(ptr, mut_span->_internal.ptr, to_copy);
+  mut_span->_internal.capacity -= to_copy;
+  mut_span->_internal.ptr += to_copy;
```

Comment:
probably is not the mut_span that needs to be updated but the pointer to user_data??

Reply:
They are one and the same.  The userdata void* pointer is cast to mut_span.